### PR TITLE
Add support for s390x

### PR DESCRIPTION
**Pull request by abhay1722**

### CODE REVIEW

**Comment by abhay1722** [comment]
Can someone please review this?

**Comment by agentzh** [comment]
@abhay1722 Thanks for your work! We do not have such arch around so it may take time on our side before we are able to review and test anything. Thanks for your understanding!

**Comment by agentzh** [comment]
@abhay1722 I suggest you post the patch to the official LuaJIT repo or mailing list for review.

**Comment by abhay1722** [comment]
@agentzh Thanks for the reply.
I will post the patch to the mailing list for review.
However, as far as the official luajit repo is concerned, we've already raised a PR for s390x support an year back, which is still open. Please refer to this [link](https://github.com/LuaJIT/LuaJIT/pull/395).
Also, the openrestry/luajit2 needs more changes than the official luajit repo.
Could you please take a look?

**Comment by agentzh** [comment]
@abhay1722 I'm not familiar with that arch myself yet, unfortunately. So it will take more time for me before I can review this. Thanks for your understanding. Thanks for your trust.

**Comment by agentzh** [comment]
@abhay1722 Have you tried running our test suite on that arch? I mean this:

https://github.com/openresty/luajit2-test-suite

File: src/lib_jit.c
Comment (agentzh) [inline src/lib_jit.c]: Oh, this is a good catch. Will you prepare a separate PR for this fix? So that we can merge this much sooner.

File: src/lj_ccall.c
Comment (agentzh) [inline src/lj_ccall.c]: I think we'd better guard this variable with the macro as well otherwise we may get C compiler warnings for variables set but not used on other architectures.

**Review by agentzh — CHANGES_REQUESTED** [review]

**Comment by agentzh** [comment]
@abhay1722 I just did a quick review above. As long as the new arch's code are guarded by the macro, maybe we can merge this thing first as long as the full test suite is passing.

**Comment by abhay1722** [comment]
@agentzh Thanks for the heads up!
I have added some new files that will be required for successfully building and testing luajit2 on s390x.
Also, the macro has been put around the onstack variable.

In Addition, I also ran the test suite on s390x using the https://github.com/openresty/luajit2-test-suite repo and all the tests passed successfully.

Please take a look and review the same.

**Comment by agentzh** [comment]
@abhay1722 Thanks for your quick update. I wonder if it's possible to run the test suite via qemu for this arch. Then we can more easily get this covered by regression testing in the future.

File: src/lib_jit.c
Comment (agentzh) [reply src/lib_jit.c]: @abhay1722 Have you missed this comment?

File: src/lj_ccall.c
Comment (agentzh) [inline src/lj_ccall.c]: Seems like you are using this variable as a boolean. The `MSize` type is confusing.

**Review by agentzh — CHANGES_REQUESTED** [review]

File: src/lib_jit.c
Comment (abhay1722) [reply src/lib_jit.c]: @agentzh No, i took a note of it.
I'll be preparing a separate PR for the same, shortly afterwards.

Comment (agentzh) [reply src/lib_jit.c]: @abhay1722 Great. Thanks.

Comment (abhay1722) [reply src/lib_jit.c]: @agentzh I have raised PR for the above changes, please refer to [this](https://github.com/openresty/luajit2/pull/51).

File: src/lj_ccall.c
Comment (abhay1722) [reply src/lj_ccall.c]: > @abhay1722 Thanks for your quick update. I wonder if it's possible to run the test suite via qemu for this arch. Then we can more easily get this covered by regression testing in the future.

@agentzh Yes it is possible to run the test suite via qemu on s390x. I am checking the same.
Will keep you posted.

**Comment by abhay1722** [comment]
> @abhay1722 Thanks for your quick update. I wonder if it's possible to run the test suite via qemu for this arch. Then we can more easily get this covered by regression testing in the future.

Yes @agentzh, it is possible to run the test suite via qemu for s390x. I am looking into the same. Will keep you posted.

File: src/lj_ccall.c
Comment (abhay1722) [reply src/lj_ccall.c]: Yes indeed, I am using the `onstack` variable as a boolean. Replaced `MSize` type with `uint32_t`.
Thanks for the input.

**Comment by kapouer** [comment]
With my debian maintainer hat on, i can tell s390x support would be nice. Also if this version of luajit gets into debian, builds and tests would be run on real s390x hardware.

File: test/test.lua
Comment (agentzh) [inline test/test.lua]: @abhay1722 I wonder if this test file has something that is not yet covered by the luajit2-test-suite repo?

**Review by agentzh — CHANGES_REQUESTED** [review]

File: test/test.lua
Comment (agentzh) [reply test/test.lua]: @abhay1722 BTW, general LuaJIT tests should go to the luajit2-test-suite repo. The test suite in this repo only exercise features specific to this (openresty's) branch of LuaJIT.

**Review by agentzh — CHANGES_REQUESTED** [review]

File: test/test.lua
Comment (abhay1722) [reply test/test.lua]: @agentzh Thanks for pointing this out.
This is just a sample test file to check the s390x assembly code added in the repo, to check simple operations like addition,loops,functions etc. It doesn't affect the mainline code of the (openresty's) branch of LuaJIT.
This test file can be removed now. Please let me know if the PR is good to merge now.

**Comment by abhay1722** [comment]
> Please let me know if the PR is good to merge now.

Hi @agentzh, Any Updates on this?

**Comment by siddhesh** [comment]
I've got the s390x patchset into my repo and in Fedora.  Tests run clean on the Fedora builders.

@agentzh would it be possible for you to merge in changes from my repo?  The v2.1 branch is here:

https://github.com/siddhesh/LuaJIT/tree/v2.1

The master branch has all that and some more changes from the openresty fork.

**Comment by agentzh** [comment]
@abhay1722 Sorry for the delay. I'm on it.

**Comment by agentzh** [comment]
@siddhesh If you have any further changes, will you post the patches here for @abhay1722 to consider or create a separate pull request? It'll be much easier for the review and merge process. Thanks!

File: src/lib_jit.c
Comment (agentzh) [reply src/lib_jit.c]: Already merged.

File: test/test.lua
Comment (agentzh) [reply test/test.lua]: @abhay1722 Fair enough.

**Comment by agentzh** [comment]
@abhay1722 It seems like this branch fails to cross-compile using Fedora's s390x gcc on x86_64:

https://gist.github.com/agentzh/0c742aca16261137e0b408618b5146f8

I'm using Fedora 28 x86_64. And the gcc toolchain is from Fedora 28's official dnf repo:

```
gcc-s390x-linux-gnu.x86_64 : Cross-build binary utilities for s390x-linux-gnu
gcc-c++-s390x-linux-gnu.x86_64 : Cross-build binary utilities for s390x-linux-gnu
binutils-s390x-linux-gnu.x86_64 : Cross-build binary utilities for s390x-linux-gnu
```

Am I doing something wrong or this branch lacks cross-compilation support? Will you have a look please? Thanks!

**Comment by agentzh** [comment]
@abhay1722 Oh, BTW, I'm using the following shell script to do the cross-compilation on Fedora 28 x86_64:

```bash
#!/bin/bash

PREFIX=/opt/luajit-s390x

sudo rm -rf $PREFIX

make clean || exit 1
make PREFIX=$PREFIX CCDEBUG=-g3 HOST_CC="gcc -g3 -O0" XCFLAGS="-DLUAJIT_ENABLE_LUA52COMPAT -I/usr/s390x-linux-gnu/include -g3 -O0" CROSS=s390x-linux-gnu- -j17 || exit 1
sudo make install PREFIX=$PREFIX
```

I've been successfully using similar scripts to cross-compile our LuaJIT in the same Fedora x86_64 system for the arm and aarch64 targets.

**Comment by siddhesh** [comment]
> @siddhesh If you have any further changes, will you post the patches here for @abhay1722 to consider or create a separate pull request? It'll be much easier for the review and merge process. Thanks!

Here you go:

https://github.com/siddhesh/LuaJIT/tree/openresty-s390x

It's from the linux-on-ibm-z project itself (I volunteered to merge in the patches for them since upstream LuaJIT has been silent) and I don't expect the patchset to be too different from that of @abhay1722 's, with the exception of a patch or two and dropping of tests since my tree had LuaJIT-test-cleanup integrated into it.

**Comment by abhay1722** [comment]
> @abhay1722 It seems like this branch fails to cross-compile using Fedora's s390x gcc on x86_64:
> 
> https://gist.github.com/agentzh/0c742aca16261137e0b408618b5146f8
> 
> I'm using Fedora 28 x86_64. And the gcc toolchain is from Fedora 28's official dnf repo:
> 
> ```
> gcc-s390x-linux-gnu.x86_64 : Cross-build binary utilities for s390x-linux-gnu
> gcc-c++-s390x-linux-gnu.x86_64 : Cross-build binary utilities for s390x-linux-gnu
> binutils-s390x-linux-gnu.x86_64 : Cross-build binary utilities for s390x-linux-gnu
> ```
> 
> Am I doing something wrong or this branch lacks cross-compilation support? Will you have a look please? Thanks!

@agentzh Thanks for replying.
Oh okay sure, I'll take a look and get back to you.

**Comment by abhay1722** [comment]
> @abhay1722 Oh, BTW, I'm using the following shell script to do the cross-compilation on Fedora 28 x86_64:
> 
> ```shell
> #!/bin/bash
> 
> PREFIX=/opt/luajit-s390x
> 
> sudo rm -rf $PREFIX
> 
> make clean || exit 1
> make PREFIX=$PREFIX CCDEBUG=-g3 HOST_CC="gcc -g3 -O0" XCFLAGS="-DLUAJIT_ENABLE_LUA52COMPAT -I/usr/s390x-linux-gnu/include -g3 -O0" CROSS=s390x-linux-gnu- -j17 || exit 1
> sudo make install PREFIX=$PREFIX
> ```
> 
> I've been successfully using similar scripts to cross-compile our LuaJIT in the same Fedora x86_64 system for the arm and aarch64 targets.

Hi @agentzh ,
I tried compiling cross compiling this s390x branch using the same script you have mentioned and it failed with similar errors as you encountered. So, I am still figuring it out.

However, I also tried cross compiling arm binaries using the script, but it did not work on my end:-
https://gist.github.com/abhay1722/15deefad54bd96ea0641ae4369b39ea6

I'm using Fedora 28 x86_64. And the gcc toolchain is from Fedora 28's official dnf repo:
`gcc-arm-linux-gnu gcc-c++-arm-linux-gnu binutils-arm-linux-gnu`

Could you tell me where I am going wrong and also what kind of `env` your using to run this script for both archs(`s390x,arm`)?

**Comment by agentzh** [comment]
@abhay1722 Hmm, I now remember that I had to use a 3rd-party yum repos for cross-compilation for ARM (and Aarch64):

https://copr.fedorainfracloud.org/coprs/lantw44/arm-linux-gnueabihf-toolchain/

Maybe it is just that Fedora's standard packages are broken...Alas.

**Comment by siddhesh** [comment]
@agentzh the fedora packages are just the compiler, assembler+linker and kernel headers but no glibc.  To do a cross build you'd first have to cross-build glibc and then use that as the sysroot to build luajit.  So it should be:

- Build glibc:

`../configure --prefix=/ --target=s390x-linux-gnu --host=s390x-linux-gnu --build=x86_64-linux-gnu --with-headers=/usr/s390-linux-gnu/include CC="s390x-linux-gnu-gcc" && make && make DESTDIR=$HOME/sysroot-s390x-linux-gnu -j install`

- Build luajit (static since dynamic setups are slightly harder to put together):

`make BUILDMODE=static HOST_CC="gcc -g3 -O0" CC="s390x-linux-gnu-gcc -DLUAJIT_ENABLE_LUA52COMPAT -I/usr/s390-linux-gnu/include -I$HOME/sysroot-s390x-linux-gnu/include --sysroot $HOME/sysroot-s390x-linux-gnu -static -g3 -O0" -j`

**Comment by agentzh** [comment]
@siddhesh Ah, thanks for the tip! How did you install the cross kernel header? I mean the `/usr/s390-linux-gnu/include` stuff. Is there an rpm package readily available? Many thanks!

**Comment by siddhesh** [comment]
Oops, forgot to mention the `kernel-cross-headers` package.

**Comment by agentzh** [comment]
@abhay1722 I'm seeing the following gcc warnings while compiling this branch:

```
In file included from lib_table.c:22:
lj_lib.h:75:36: warning: ‘lj_cf_table_pack’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_table.c:313:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(table_pack)
 ^~~~~~~~
In file included from lib_debug.c:20:
lj_lib.h:75:36: warning: ‘lj_cf_debug_setuservalue’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_debug.c:271:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(debug_setuservalue)
 ^~~~~~~~
lj_lib.h:75:36: warning: ‘lj_cf_debug_getuservalue’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_debug.c:260:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(debug_getuservalue)
 ^~~~~~~~
In file included from lib_base.c:37:
lj_lib.h:75:36: warning: ‘lj_cf_rawlen’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_base.c:209:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(rawlen)  LJLIB_REC(.)
 ^~~~~~~~
CC        lib_ffi.o
...
In file included from lib_table.c:22:
lj_lib.h:75:36: warning: ‘lj_cf_table_pack’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_table.c:313:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(table_pack)
 ^~~~~~~~
In file included from lib_debug.c:20:
lj_lib.h:75:36: warning: ‘lj_cf_debug_setuservalue’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_debug.c:271:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(debug_setuservalue)
 ^~~~~~~~
lj_lib.h:75:36: warning: ‘lj_cf_debug_getuservalue’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_debug.c:260:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(debug_getuservalue)
 ^~~~~~~~
s390x-linux-gnu-gcc -g3 -O2 -fomit-frame-pointer -Wall   -DLUAJIT_ENABLE_LUA52COMPAT -DLUA_USE_APICHECK -DLUA_USE_ASSERT -I/usr/s390-linux-gnu/include -I/usr/s390x-linux-gnu/sys-root/include -g3 -O0   -c -o lib_string.o lib_string.c
s390x-linux-gnu-gcc -g3 -O2 -fomit-frame-pointer -Wall   -DLUAJIT_ENABLE_LUA52COMPAT -DLUA_USE_APICHECK -DLUA_USE_ASSERT -I/usr/s390-linux-gnu/include -I/usr/s390x-linux-gnu/sys-root/include -g3 -O0   -c -o lib_base.o lib_base.c
In file included from lib_base.c:37:
lj_lib.h:75:36: warning: ‘lj_cf_rawlen’ defined but not used [-Wunused-function]
 #define LJLIB_CF(name)  static int lj_cf_##name(lua_State *L)
                                    ^~~~~~
lib_base.c:209:1: note: in expansion of macro ‘LJLIB_CF’
 LJLIB_CF(rawlen)  LJLIB_REC(.)
 ^~~~~~~~
```

Will you please address them? Thanks!

**Comment by agentzh** [comment]
@siddhesh Thanks for your help! I now successfully cross-built LuaJIT with the s390x gcc and glibc. Unfortunately, qemu-s390x always crashes with segmentation faults when it tries to run the resulting luajit binary:

```
Thread 1 "qemu-s390x" received signal SIGSEGV, Segmentation fault.
0x000055555597f32e in static_code_gen_buffer ()
(gdb) bt
#0  0x000055555597f32e in static_code_gen_buffer ()
#1  0x00005555555ee33a in cpu_tb_exec (itb=<optimized out>, itb=<optimized out>, cpu=<optimized out>) at /usr/src/debug/qemu-2.11.2-5.fc28.x86_64/accel/tcg/cpu-exec.c:167
#2  cpu_loop_exec_tb (tb_exit=<synthetic pointer>, last_tb=<synthetic pointer>, tb=<optimized out>, cpu=<optimized out>) at /usr/src/debug/qemu-2.11.2-5.fc28.x86_64/accel/tcg/cpu-exec.c:627
#3  cpu_exec (cpu=<optimized out>) at /usr/src/debug/qemu-2.11.2-5.fc28.x86_64/accel/tcg/cpu-exec.c:736
#4  0x00005555555f2dd0 in cpu_loop (env=0x555557a06900) at /usr/src/debug/qemu-2.11.2-5.fc28.x86_64/linux-user/main.c:3255
#5  0x00005555555c5b32 in main (argc=<optimized out>, argv=0x7fffffffcb58, envp=<optimized out>) at /usr/src/debug/qemu-2.11.2-5.fc28.x86_64/linux-user/main.c:4882
```

Maybe it's an issue in Fedora 28's qemu-s390x package...Alas.

**Comment by agentzh** [comment]
@siddhesh OK, never mind, the qemu-s390x crashes were due to the dynamic library search path misconfiguration on my side. I had to set the following env to make it work:

```
export QEMU_SET_ENV=LD_LIBRARY_PATH=/usr/s390-linux-gnu/lib:/usr/s390x-linux-gnu/sys-root/lib:$HOME/git/luajit2-test-suite/s390x
```

Now most of the tests are finally passing on my side, but I'm still seeing some notable failures:

https://gist.github.com/agentzh/de31ae0b29c56c0392de68aeb9188977

There are 20 test failures. The failures due to the error message `sh: error while loading shared libraries: /usr/s390x-linux-gnu/sys-root/lib/libdl.so.2: ELF file data encoding not little-endian` can be ignored for now since it's apparently my own qemu environment problems. Those segmentation faults, illegal instruction exceptions, and assertion failures would need your eyes @abhay1722. Are you seeing them on your side?

**Comment by abhay1722** [comment]
> @siddhesh OK, never mind, the qemu-s390x crashes were due to the dynamic library search path misconfiguration on my side. I had to set the following env to make it work:
> 
> ```
> export QEMU_SET_ENV=LD_LIBRARY_PATH=/usr/s390-linux-gnu/lib:/usr/s390x-linux-gnu/sys-root/lib:$HOME/git/luajit2-test-suite/s390x
> ```
> 
> Now most of the tests are finally passing on my side, but I'm still seeing some notable failures:
> 
> https://gist.github.com/agentzh/de31ae0b29c56c0392de68aeb9188977
> 
> There are 20 test failures. The failures due to the error message `sh: error while loading shared libraries: /usr/s390x-linux-gnu/sys-root/lib/libdl.so.2: ELF file data encoding not little-endian` can be ignored for now since it's apparently my own qemu environment problems. Those segmentation faults, illegal instruction exceptions, and assertion failures would need your eyes @abhay1722. Are you seeing them on your side?

@agentzh @siddhesh Sorry for replying late. 
I was trying to cross-compile this branch on fedora x86_64 myself. I was successfully able to build and install glibc, however I am encountering issues while building luajit. Could you please provide exact steps to successfully build the same, so that I can move on to checking the test case failures.

**Comment by abhay1722** [comment]
> > @siddhesh OK, never mind, the qemu-s390x crashes were due to the dynamic library search path misconfiguration on my side. I had to set the following env to make it work:
> > ```
> > export QEMU_SET_ENV=LD_LIBRARY_PATH=/usr/s390-linux-gnu/lib:/usr/s390x-linux-gnu/sys-root/lib:$HOME/git/luajit2-test-suite/s390x
> > ```
> > 
> > 
> > Now most of the tests are finally passing on my side, but I'm still seeing some notable failures:
> > https://gist.github.com/agentzh/de31ae0b29c56c0392de68aeb9188977
> > There are 20 test failures. The failures due to the error message `sh: error while loading shared libraries: /usr/s390x-linux-gnu/sys-root/lib/libdl.so.2: ELF file data encoding not little-endian` can be ignored for now since it's apparently my own qemu environment problems. Those segmentation faults, illegal instruction exceptions, and assertion failures would need your eyes @abhay1722. Are you seeing them on your side?
> 
> @agentzh @siddhesh Sorry for replying late.
> I was trying to cross-compile this branch on fedora x86_64 myself. I was successfully able to build and install glibc, however I am encountering issues while building luajit. Could you please provide exact steps to successfully build the same, so that I can move on to checking the test case failures.

@agentzh Could you please update on this?

**Comment by agentzh** [comment]
@abhay1722 Sorry for the late reply. This is the commands I used to build your LuaJIT branch:

```bash
PREFIX=/opt/luajit-s390x

sudo rm -rf $PREFIX

make clean || exit 1
make Q= PREFIX=$PREFIX CCDEBUG=-g3 HOST_CC="gcc -g3 -O0" TARGET_XCFLAGS="-DLUAJIT_ENABLE_LUA52COMPAT -DLUA_USE_APICHECK -DLUA_USE_ASSERT -I/usr/s390-linux-gnu/include -I/usr/s390x-linux-gnu/sys-root/include -g3 -O0" CROSS=s390x-linux-gnu- -j17 || exit 1
sudo make install PREFIX=$PREFIX
```

**Comment by abhay1722** [comment]
@agentzh Do ya'll have a Jenkins CI Infrastructure where continuous builds are executed for openresty/luajit2.
Is it be possible for us to add our s390x systems on the luajit2 project's CI?

**Comment by siddhesh** [comment]
> @agentzh Do ya'll have a Jenkins CI Infrastructure where continuous builds are executed for openresty/luajit2.
> Is it be possible for us to add our s390x systems on the luajit2 project's CI?

Ugh, very sorry I should have suggested this earlier...

I got access to an s390x machine for LuaJIT development from the Linux Foundation.  I'll be happy to set up CI for luajit2 there and also give access to @agentzh for verification so that we get out of this cross building madness.  Please send me your public key and I'll set up an account for you.

**Comment by agentzh** [comment]
@siddhesh That sounds awesome :) Thanks a lot!

**Comment by siddhesh** [comment]
@agentzh please email me your ssh public key.

**Comment by abhay1722** [comment]
@siddhesh Thanks alot for the support!
@agentzh Please let me know if the build is successful on s390x machine.

**Comment by agentzh** [comment]
@abhay1722 But I still think warnings from new gcc compilers (8.1.1) are worth fixing, no? Have you got any luck using my commands to do the cross-compiling?

**Comment by agentzh** [comment]
@abhay1722 I mean the gcc warnings provided here:

https://github.com/openresty/luajit2/pull/50#issuecomment-497876563

**Comment by abhay1722** [comment]
> 
> 
> @abhay1722 But I still think warnings from new gcc compilers (8.1.1) are worth fixing, no? Have you got any luck using my commands to do the cross-compiling?

@agentzh I did try the steps you mentioned in this [comment](https://github.com/openresty/luajit2/pull/50#issuecomment-507467910), however, I am encountering the following issues:-

https://gist.github.com/abhay1722/92a67cae41cda258dfd99da8456feca3

Please let me know where I could be going wrong.

**Comment by abhay1722** [comment]
@agentzh Although, I would still suggest you to try this branch on an s390x machine and check if such warnings occur there too.

**Comment by agentzh** [comment]
@abhay1722 Re https://gist.github.com/abhay1722/92a67cae41cda258dfd99da8456feca3

Seems like you failed to specify your cross-compiled glibc's include directory?

OK, I'll merge this first.

**Comment by agentzh** [comment]
@abhay1722 I think those warnings are only reproducible when you specify the build option `-DLUAJIT_ENABLE_LUA52COMPAT`. Without this option, there is no warnings. Will you confirm?

I've already merged this PR and the remaining warnings should be addressed in another PR :) Thanks!

**Comment by abhay1722** [comment]
> @abhay1722 I think those warnings are only reproducible when you specify the build option -DLUAJIT_ENABLE_LUA52COMPAT. Without this option, there is no warnings. Will you confirm?
> I've already merged this PR and the remaining warnings should be addressed in another PR :) Thanks!

@agentzh Yes, you're correct, the warnings only occur with the `-DLUAJIT_ENABLE_LUA52COMPAT` build option.
Thanks!!

**Pull request closed without merging**
